### PR TITLE
Preparation for goog.module transition: base.js, deps.js

### PR DESCRIPTION
**Pull request by cpcallen**
## The basics

- ~~[ ] I branched from develop~~
- ~~[ ] My pull request is against develop~~
- [X] I branched from goog_module
- [X] My pull request is against goog_module
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Proposed Changes

* Use the current standard Closure Library version of `closure/goog/base.js`.
* In uncompiled mode, load dependency graph from `tests/deps.js`.
  * New gulp task `buildDeps` (`npm run build:deps`) to create `tests/deps.js`.
  * `blockly_uncompressed.js` is now handwritten.
    * And simplified; in particular, BLOCKLY_BOOT is gone.
    * And linted!
  * Old gulp task `buildUncompressed` is deleted.
* For consistency with the Closure documentation and base.js, consistently refer to what blockly_uncompressed.js is used for as "uncompiled mode" rather than "uncompressed mode" (but don't yet rename it to blockly_uncompiled.js).
* Have `eslint` ignore "undefined" global variable `exports`, used to specify exports from `goog.module` modules.

### Reason for Changes

The version of base.js we have been using is a cut-down copy of an older version, stripped of everything that wasn't strictly needed for Blockly.  We now need things that were previously stripped out, so restore the current version from https://github.com/google/closure-library/blob/master/closure/goog/base.js.  We can always strip out unused sections again in future - once we know which sections those are.

Loading dependency graph info from `tests/deps.js` (instead of having it embedded in `blockly_uncompressed.js` helps maintain a clear separation between generated and handwritten code, avoids having code duplicated between the generator script and its output, and helped find some actual errors the handwritten parts because they can now be linted.  It also more closely follows the Closure Library convention.

### Additional Information

This PR (or another to the same effect) must be merged to the `goog_module` branch before any others that actually  carrying out `goog.provides` -> `goog.module` migration.

### CODE REVIEW

File: core/contextmenu_items.js
Comment (rachel-fenichel) [inline core/contextmenu_items.js:194]: This disables the rule for the rest of the file. Use `eslint-disable-line` or `eslint-disable-next-line` instead.

https://eslint.org/docs/user-guide/configuring/rules#disabling-rules

**Review by rachel-fenichel — APPROVED** [review]
Approved after lint fix.

**Comment by cpcallen** [comment]
Blocked on #5018.

File: core/contextmenu_items.js
Comment (cpcallen) [reply core/contextmenu_items.js]: TIL.  Ta!

**Pull request merged**
